hypex/dataset/dataset.py

```diff
@@ -0,0 +1,274 @@
+        >>> binary_outcome = False # If outcome should be binary
+        >>> na_columns = ['feature_col_1', 'feature_col_2'] # List of columns with NA values
+        >>> na_step = 15 # Number of period to make NaN (step of range)
+        >>>
+        >>> sample_data = Dataset(num_records=num_records,
+        >>>                       num_main_causes_cols=num_main_causes_cols,
+        >>>                       num_info_cols=num_info_cols,
+        >>>                       num_treatments=num_treatments,
+        >>>                       num_outcomes=num_outcomes,
+        >>>                       binary_outcome=binary_outcome,
+        >>>                       na_columns=na_columns,
+        >>>                       na_step=na_step)
+        >>>
+        >>> print(sample_data.df)
+        >>>
+        >>> # Without treatment
+        >>> sample_data = Dataset(num_treatments=0)
+        >>>
+        >>> print(sample_data.df)
+        >>>
+        >>>
+        >>> print(sample_data) # More details about attributes
+    """
+    def __init__(
+            self,
+            num_records: int = 5000,
+            num_main_causes_cols: int = 4,
+            num_info_cols: int = 2,
+            num_treatments: int = 1,
+            num_discrete_main_causes_cols: int = 1,
+            num_outcomes: int = 1,
+            binary_outcome: bool = False,
+            na_columns: str | list = '',
+            na_step: int = 0) -> None:
+        """
+        Initialize the Dataset object.
+
+        Args:
+            num_records:
+                Number of records in created dataset. Defaults to 5000
+            num_main_causes_cols:
+                Number of columns correlating with outcome. Defaults to 4
+            num_info_cols:
+                Number of columns with information about each record. Defaults to 2
+            num_treatments:
+                Number of treatments (info about 'treatment' 0 or 1). Defaults to 1
+            num_discrete_main_causes_cols:
+                Number of discrete columns correlating with outcome. Defaults to 1
+            num_outcomes:
+                Number of columns with target values. Defaults to 1
+            binary_outcome:
+                If outcome should be binary. Defaults to False
+            na_columns:
+                Column or list of columns with NA values. Defaults to None
+            na_step:
+                Number of period to make NaN (step of range). Defaults to 0
+        """
+        self.num_records = num_records
+        self.num_main_causes_cols = num_main_causes_cols
+        self.num_info_cols = num_info_cols
+        self.num_treatments = num_treatments
+        self.num_outcomes = num_outcomes
+        self.num_discrete_main_causes_cols = num_discrete_main_causes_cols
+        self.binary_outcome = binary_outcome
+        self.na_columns = na_columns
+        self.na_step = na_step
+        self.treatment_name: str | list = ''
```

Comment:
Use library typing for this. 
`import typing as tp
tp.Union[str, list]`

This is to keep python 3.8 support intact